src/libraries/System.Collections.Immutable/ref/System.Collections.Immutable.cs

```diff
@@ -223,24 +344,33 @@ public sealed partial class Builder : System.Collections.Generic.ICollection<T>,
             public bool Contains(T item) { throw null; }
             public void CopyTo(T[] array, int index) { }
             public System.Collections.Generic.IEnumerator<T> GetEnumerator() { throw null; }
+            [System.Diagnostics.Contracts.PureAttribute]
             public int IndexOf(T item) { throw null; }
+            [System.Diagnostics.Contracts.PureAttribute]
             public int IndexOf(T item, int startIndex) { throw null; }
+            [System.Diagnostics.Contracts.PureAttribute]
             public int IndexOf(T item, int startIndex, int count) { throw null; }
+            [System.Diagnostics.Contracts.PureAttribute]
             public int IndexOf(T item, int startIndex, int count, System.Collections.Generic.IEqualityComparer<T>? equalityComparer) { throw null; }
             public void Insert(int index, T item) { }
 #if !NETSTANDARD1_0
             public ref readonly T ItemRef(int index) { throw null; }
 #endif
+            [System.Diagnostics.Contracts.PureAttribute]
             public int LastIndexOf(T item) { throw null; }
+            [System.Diagnostics.Contracts.PureAttribute]
             public int LastIndexOf(T item, int startIndex) { throw null; }
+            [System.Diagnostics.Contracts.PureAttribute]
             public int LastIndexOf(T item, int startIndex, int count) { throw null; }
+            [System.Diagnostics.Contracts.PureAttribute]
             public int LastIndexOf(T item, int startIndex, int count, System.Collections.Generic.IEqualityComparer<T>? equalityComparer) { throw null; }
             public System.Collections.Immutable.ImmutableArray<T> MoveToImmutable() { throw null; }
             public bool Remove(T element) { throw null; }
             public void RemoveAt(int index) { }
             public void Reverse() { }
             public void Sort() { }
             public void Sort(System.Collections.Generic.IComparer<T>? comparer) { }
+            [System.Diagnostics.Contracts.PureAttribute]
             public void Sort(System.Comparison<T> comparison) { }
```

Comment:
@AArnott how is it that this method is Pure?  Isn't it changing the state of the object in the same way other Sort methods are?

Reply:
Ya, that's a bug.